### ISSUE TITLE
Add supporting of urllib3 v2

### ISSUE THREAD

**Issue by ALeksandr-Fuze** [python-poetry/poetry#8178]
- [+] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [+] I have searched the [FAQ](https://python-poetry.org/docs/faq/) and general [documentation](https://python-poetry.org/docs/) and believe that my question is not already covered.

## Feature Request
The latest version of packages required urllib3 v2 when the poetry requirement is urllib3 V1. So after adding any package with this requirement, I get `HTTPResponse.__init__() got an unexpected keyword argument 'strict'` when updating poetry to the last version using **pip** or **poetry self update**.

**Comment by dimbleby** [comment]
duplicate #7878 etc, please close 

tldr poetry declares its dependencies correctly so if you are seeing problems it is because you haven't installed poetry right
